moveit_planners/ompl/ompl_interface/src/parameterization/work_space/pose_model_state_space.cpp

```diff
@@ -131,9 +133,31 @@ void PoseModelStateSpace::sanityChecks() const
 void PoseModelStateSpace::interpolate(const ompl::base::State* from, const ompl::base::State* to, const double t,
                                       ompl::base::State* state) const
 {
-  // interpolate in joint space
+  // we want to interpolate in Cartesian space to avoid rejection of path constraints
+
+  // interpolate in joint space to find a suitable seed for IK
   ModelBasedStateSpace::interpolate(from, to, t, state);
-  computeStateFK(state);
+  double d_joint = ModelBasedStateSpace::distance(from, state);
+
+  // interpolate SE3 components
+  for (std::size_t i = 0; i < poses_.size(); ++i)
+  {
+    poses_[i].state_space_->interpolate(from->as<StateType>()->poses[i], to->as<StateType>()->poses[i], t,
+                                        state->as<StateType>()->poses[i]);
+  }
+
+  // the call above may reset all flags for state; but we know the pose we want flag should be set
```

Comment:
This comment is a bit confusing for me understand. Should this say "we know the flag for the pose we want should be set"?

And what flag? The "pose computed" flag? Might help to be more specific